### PR TITLE
[T5A1][T10-B2] Li Zhiyuan

### PR DESCRIPTION
**Pull request by lzyuan168**
user story, use case, non-functional requirements

### CODE REVIEW

**Comment by lzyuan168** [comment]
ready for review

File: doc/DeveloperGuide.md [hunk 1]
```diff
@@ -73,13 +75,44 @@ Use case ends.
...
 > 3a1. AddressBook shows an error message <br>
   Use case resumes at step 2
+  
+#### Use case: Rename tag
+
+**MSS**
+
+1. User requests to list persons
+2. AddressBook shows a list of persons
+3. User requests to rename a specific person in the list
```
Comment (nirandiw) [inline doc/DeveloperGuide.md]: Step missing: Address book's action

File: doc/DeveloperGuide.md [hunk 2]
```diff
@@ -73,13 +75,44 @@ Use case ends.
 
 > 3a1. AddressBook shows an error message <br>
   Use case resumes at step 2
+  
+#### Use case: Rename tag
+
+**MSS**
```
Comment (nirandiw) [inline doc/DeveloperGuide.md]: - The use case was to rename a tag not edit a tag of a specific person. 
- Keep in mind in use-cases it is important to go back and forth from the user to the system

File: doc/DeveloperGuide.md [hunk 3]
```diff
@@ -73,13 +75,44 @@ Use case ends.
...
+5. AddressBook confirms edit with user
+6. User confirms edit
+7. AddressBook edits the tag of the person <br>
+Use case ends.
+
+**Extensions**
+
+2a. The list is empty
+
+> Use case ends
```
Comment (nirandiw) [inline doc/DeveloperGuide.md]: Shouldn't the system inform the user that the list is empty

File: doc/DeveloperGuide.md [hunk 4]
```diff
@@ -73,13 +75,44 @@ Use case ends.
...
+> 7a1. AddressBook shows an error message <br>
+  Use case resumes at step 4
+  
 ## Appendix C : Non Functional Requirements
 
 1. Should work on any [mainstream OS](#mainstream-os) as long as it has Java 8 or higher installed.
 2. Should be able to hold up to 1000 persons.
 3. Should come with automated unit tests and open source code.
 4. Should favor DOS style commands over Unix-style commands.
+5. Should be user friendly and present data in easy readable manner.
```
Comment (nirandiw) [inline doc/DeveloperGuide.md]: - Try to make the NFRs more precise and verifiable. 
i.e. what is an "easy readable manner"?
what is user-friendly and to what kind of user should it be user-friendly?

**Comment by nirandiw** [comment]
@lzyuan168 Some comments added. Please close the PR after reading comments.

**Pull request closed without merging**